src/main/java/com/truelayer/java/http/auth/cache/SimpleCredentialsCache.java

```diff
@@ -70,4 +61,10 @@ public static class AccessTokenRecord {
         private final AccessToken token;
         private final LocalDateTime expiresAt;
     }
+
+    private static int computeCacheKey(RequestScopes scopes) {
+        List<String> sortedScopes = new ArrayList<>(scopes.getScopes());
+        Collections.sort(sortedScopes);
+        return sortedScopes.hashCode();
```

Comment:
perhaps we could even create a more human readable key, by joining all elements with a `-` delimiter. `payment-signupplus` would be a cache key example.